crates/re_viewer_context/src/time_control.rs

```diff
@@ -375,47 +379,53 @@ impl TimeControl {
 
     /// playback fps
     pub fn fps(&self) -> Option<f32> {
-        self.states.get(&self.timeline).map(|state| state.fps)
+        self.states.get(self.timeline.get()).map(|state| state.fps)
     }
 
     /// playback fps
     pub fn set_fps(&mut self, fps: f32) {
-        if let Some(state) = self.states.get_mut(&self.timeline) {
+        if let Some(state) = self.states.get_mut(self.timeline.get()) {
             state.fps = fps;
         }
     }
 
     /// Make sure the selected timeline is a valid one
     pub fn select_a_valid_timeline(&mut self, times_per_timeline: &TimesPerTimeline) {
-        for timeline in times_per_timeline.timelines() {
-            if &self.timeline == timeline {
-                return; // it's valid
+        fn is_timeline_valid(selected: &Timeline, times_per_timeline: &TimesPerTimeline) -> bool {
+            for timeline in times_per_timeline.timelines() {
+                if selected == timeline {
+                    return true; // it's valid
+                }
             }
+            false
         }
-        if let Some(timeline) = default_timeline(times_per_timeline.timelines()) {
-            self.timeline = *timeline;
-        } else {
-            self.timeline = Default::default();
+
+        if !self.timeline.is_auto() && is_timeline_valid(self.timeline.get(), times_per_timeline) {
+            return;
```

Comment:
So if the auto is invalid, we still keep it? That sounds wrong?

Reply:
no, if auto is invalid then this becomes (false && false) == false, so we don't return and set a new auto value.
Let me clean this up - the statement below is small enough to not make this early-out style.

Reply:
no, if auto is invalid then this becomes `(false && false) == false`, so we *don't* return and set a new auto value.
Let me clean this up - the statement below is small enough to not make this early-out style.